agent/config/builder.go

```diff
@@ -223,8 +227,7 @@ func (b *builder) sourcesFromPath(path string, format string) ([]Source, error)
 
 	if !fi.IsDir() {
 		if !shouldParseFile(path, format) {
-			b.warn("skipping file %v, extension must be .hcl or .json, or config format must be set", path)
-			return nil, nil
+			return nil, fmt.Errorf("file %v has unknown extension; must be .hcl or .json, or config format must be set", path)
```

Comment:
Will this change break the current behavior? Say, if user runs `consul agent -config-dir=/some-config-dir`. Under this `/some-config-dir` directory, there are some non-hcl/json file. Before this change, consul agent can proceed with the warning message, but this PR may stop process from being started.

Reply:
This section is only for files, not directories. The directory logic starts at https://github.com/hashicorp/consul/blob/068cb2263ab52bb1947dce5e3d23d249e09317a6/agent/config/builder.go#L240, and remains the same (skip and warn).

But technically yes, it does break current behavior. I'm arguing current behavior is incorrect though: if I run `consul agent -config-file=config.badextension`, that should fail, not skip.

Reply:
Got it. It makes sense. Thanks.